inst/@symfun/mtimes.m

```diff
@@ -58,7 +58,8 @@
 
 function h = mtimes(f, g)
   [vars, s1, s2] = helper_symfun_binops(f, g);
-  h = symfun(s1 * s2, vars);
+  res = s1 * s2;
+  h = symfun(res, symvar(res));
```

Comment:
I can't find a reference but IIRC, SMT did this too.  I agree it does seem reasonable to take the union of the inputs as the vars of the result.  But anyway this has little to do with this PR right?

>The result will definitely have variables from both the symbols involved in that operation

Hmm, but this may not be the correct result.  Consider:
````
octave:9> f(x,y) = 2*x
f(x, y) = (symfun) 2⋅x
octave:10> g(x,y) = 1
g(x, y) = (symfun) 1
octave:11> f*g
ans(x) = (symfun) 2⋅x
````
(Previous to this PR, that would be `ans(x,y)`.)

>  Also, the variables of the symfun are only considered as final variables in any operation!!

Not sure what you mean.